base/c.jl

```diff
@@ -43,6 +43,29 @@ end
 
 typealias Coff_t FileOffset
 
+# C NUL-terminated string pointers; these can be used in ccall
+# instead of Ptr{Cchar} and Ptr{Cwchar_t}, respectively, to enforce
+# a check for embedded NUL chars in the string (to avoid silent truncation).
+immutable Cstring; p::Ptr{Cchar}; end
+immutable Cwstring; p::Ptr{Cwchar_t}; end
```

Comment:
No, I'm getting `ErrorException("invalid number of bits in type Cstring")` ... does the 2nd argument to `bitstype` have to be a numeric literal?